it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/TrustUtrControllerISpec.scala

```diff
@@ -0,0 +1,182 @@
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{Ctutr, Sautr}
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, StorageStub}
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ComponentSpecHelper
+import uk.gov.hmrc.minorentityidentificationfrontend.views.trustViews.TrustCaptureUtrViewTests
+
+class TrustUtrControllerISpec extends ComponentSpecHelper with AuthStub with StorageStub with TrustCaptureUtrViewTests {
+
+  "GET /sa-utr" should {
+    lazy val result = {
+      await(insertJourneyConfig(
+        journeyId = testJourneyId,
+        internalId = testInternalId,
+        testTrustsJourneyConfig(businessVerificationCheck = true)
+      ))
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+      get(s"/identify-your-trust/$testJourneyId/sa-utr")
+    }
+
+    "return OK" in {
+      result.status mustBe OK
+    }
+
+    "return a view which" should {
+      testCaptureUtrView(result)
+    }
+
+    "redirect to sign in page" when {
+      "the user is UNAUTHORISED" in {
+        stubAuthFailure()
+        lazy val result: WSResponse = get(s"/identify-your-trust/$testJourneyId/sa-utr")
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri("/bas-gateway/sign-in" +
+            s"?continue_url=%2Fidentify-your-trust%2F$testJourneyId%2Fsa-utr" +
+            "&origin=minor-entity-identification-frontend"
+          )
+        )
+      }
+    }
+  }
+
+  "POST /sa-utr" when {
+    "the utr is correctly formatted" should {
+      "redirect to enter your postcode" in {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          internalId = testInternalId,
+          testTrustsJourneyConfig(businessVerificationCheck = true)
+        ))
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubStoreUtr(testJourneyId, Sautr(testUtr))(OK)
+
+        lazy val result = post(s"/identify-your-trust/$testJourneyId/sa-utr")("utr" -> testUtr)
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri(routes.CaptureSaPostcodeController.show(testJourneyId).url)
+        )
+      }
+    }
+
+    "the utr is a ctutr" should {
+      "redirect to check your answers" in {
+        val testCtutr = "1234529999"
+
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          internalId = testInternalId,
+          testTrustsJourneyConfig(businessVerificationCheck = true)
+        ))
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubStoreUtr(testJourneyId, Ctutr(testCtutr))(OK)
+
+        lazy val result = post(s"/identify-your-trust/$testJourneyId/sa-utr")("utr" -> testCtutr)
+
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri(routes.CaptureSaPostcodeController.show(testJourneyId).url)
+        )
+      }
+    }
```

Comment:
This is specifically for an sautr page so I don't think this test is needed. Also there's an unused important at the top